### ISSUE TITLE
Resolve references/{id}/bases method definition ambiguity

### ISSUE THREAD

**Issue by macieksmuga** [ga4gh/schemas#415]
The current `references/{id}/bases` in `referencemethods.avdl` is a `GET` method
that includes a `ListReferenceBasesRequest` object - I’m not sure how these two
concepts can play together, as JSON objects can’t be sent to the server via a `GET` request.

I propose that this method be changed to a `POST` request to allow for paging (controlled by the server’s internally defined response byte size limit, to prevent arbitrary-size returns). 

It would go something like this, dropping the faux-REST style it’s currently wearing:

```
/****************  /references/bases  *******************/
/**
The request object for `POST /references/bases`.
*/
record ListReferenceBasesRequest {

  /**
  The reference ID whose bases are being requested
  */
  string referenceId;

  /**
  The start position (0-based) of this query. Defaults to 0.
  Genomic positions are non-negative integers less than reference length.
  Requests spanning the join of circular genomes are represented as
  two requests one on each side of the join (position 0).
  */
  long start = 0;

  /**
  The end position (0-based, exclusive) of this query. Defaults
  to the length of the requested `Reference`.
  */
  union { null, long } end = null;

  /**
  The continuation token, which is used to page through large result sets.
  To get the next page of results, set this parameter to the value of
  `nextPageToken` from the previous response, while keeping the other parameters
  identical.
  */
  union { null, string } pageToken = null;
}

/** The response from `POST /references/bases` expressed as JSON. */
record ListReferenceBasesResponse {
  /**
  The offset position (0-based) of the given sequence from the start of this
  `Reference`. This value will differ for each page in a paginated request.
   */
  long offset = 0;

  /**
  A substring of the bases that make up this reference. Bases are represented
  as IUPAC-IUB codes; this string matches the regexp `[ACGTMRWSYKVHDBN]*`.
  */
  string sequence;

  /**
  The continuation token, which is used to page through large result sets.
  Provide this value along with the original request parameters
  in a subsequent request to return the next page of
  results. This field will be empty if there aren't any additional results.
  */
  union { null, string } nextPageToken = null;
}

/**
Lists `Reference` bases by ID and optional range.
`POST /references/bases` will return a JSON version of
`ListReferenceBasesResponse`.
*/
ListReferenceBasesResponse getReferenceBases(
  ListReferenceBasesRequest request) throws GAException;
}
```

This issue is similar in flavor to #414 as it aims to nail down the semantics of paging for this protocol.

**Comment by calbach** [comment]
By convention, the equivalent mechanism to passing JSON parameters in POST is to use query parameters in GET. This is how most REST APIs work as far as I know, though I guess this isn't actually stated anywhere in the schemas specification.

e.g. `/references/123/bases?start=100&end=200&pageToken=asdf`

**Comment by diekhans** [comment]
turning this into a documentation ticket

**Comment by jeromekelleher** [comment]
ListReferenceBases is a bit of a wart --- it is different to other queries, and there doesn't seem to be any good reason for it. Are there any arguments in favour of it being a GET rather than POST?

**Comment by calbach** [comment]
Arguably all of our 'Search' methods should actually be GETs, which is the more standard way of enumerating a collection in a REST API. I believe the original reason we proposed POST for those methods was due to concerns around URL length limitations (which you are in danger of hitting when all parameters are in the ), which in hindsight were likely unwarranted.

I guess the ship has sailed on that though, so for the sake of consistency I'm not opposed to changing this to POST - though I do believe this method is more natural as a GET, GA4GH consistency aside.
